dwave_networkx/drawing/draw_parallel_embeddings.py

```diff
@@ -0,0 +1,242 @@
+        S: The optional source graph (if None, edges are colored based on node colors).
+        one_to_iterable: If True, a single source node maps to multiple target nodes.
+        node_color_dict: The node color dictionary to reference for edge coloring.
+
+    Returns:
+        edge_color_dict: A dictionary mapping each edge in G to either an embedding index or NaN.
+    """
+    if S is not None:
+        # Edges corresponding to source graph embeddings
+        edge_color_dict = {
+            (tu, tv): idx
+            for idx, emb in enumerate(embeddings)
+            for u, v in S.edges()
+            if u in emb and v in emb
+            for tu in (emb[u] if one_to_iterable else [emb[u]])
+            for tv in (emb[v] if one_to_iterable else [emb[v]])
+            if G.has_edge(tu, tv)
+        }
+
+        if one_to_iterable:
+            # Add chain edges
+            for idx, emb in enumerate(embeddings):
+                for chain in emb.values():
+                    Gchain = G.subgraph(chain)
+                    edge_color_dict.update({e: idx for e in Gchain.edges()})
+    else:
+        # If no source graph, color edges where both endpoints share the same embedding color
+        edge_color_dict = {
+            (v1, v2): node_color_dict[v1]
+            for v1, v2 in G.edges()
+            if node_color_dict[v1] == node_color_dict[v2]
+        }
+
+    return edge_color_dict
+
+
+def draw_parallel_embeddings(
+    G: nx.Graph,
+    embeddings: List[dict],
+    S: nx.Graph = None,
+    one_to_iterable: bool = True,
+    shuffle_colormap: bool = True,
+    seed: Optional[int] = None,
+    use_plt=True,
+    **kwargs,
+) -> Tuple[dict, dict]:
+    """Visualizes the embeddings using dwave_networkx's layout functions.
+
+    Args:
+        G: The target graph to be visualized.
+        embeddings: A list of embeddings.
+        S: The source graph to visualize (optional).
+        one_to_iterable: If True, allow multiple target nodes per source node,
+            values of the embedding are iterables on nodes of G.
+            If False, embedding values should be nodes of G.
+        shuffle_colormap: If True, randomize the colormap assignment.
+        seed: A seed for the random number generator.
+        **kwargs: Additional keyword arguments for the drawing functions.
+
+    Examples
+    --------
+    This example plots 3 embeddings of a length-8 path on Chimera[m=2], with
+    no embedding in the bottom right.
+
+    >>> import networkx as nx
+    >>> import dwave_networkx as dnx
+    >>> import matplotlib.pyplot as plt   # doctest: +SKIP
+    >>> G = dnx.chimera_graph(2)
+    >>> S = nx.from_edgelist({(i,i+1) for i in range(7)})
+    >>> emb = {i: 2 * (i // 2) + 4*(i % 2) for i in range(8)}  # Top-left embedding
+    >>> embs = [{k:v+8*offset for k,v in emb.items()} for offset in range(3)]
+    >>> dnx.draw_parallel_embeddings(G)    # doctest: +SKIP
```

Comment:
```suggestion
    >>> dnx.draw_parallel_embeddings(G, embs)    # doctest: +SKIP
```

Reply:
Although this doesn't work either; can't iterate through the values of an embedding since they're ints.